migrations/20201111_api_date.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * Fix dates in the database that were stored as $type string instead of Date
+ */
+/* eslint-disable no-console */
+
+const MIGRATION_NAME = '20201111_api_date';
+
+import * as Tasks from '../website/server/models/task';
+
+const progressCount = 1000;
+let count = 0;
+
+async function updateUser (user) {
+  count++;
+
+  let set = {};
+  let newDate = new Date(user.date);
+  if(isValidDate(newDate)) {
+    set = {
+      'date': newDate
+    }
+  } else {
+    set = {
+      'date': ''
+    }
+  }
+
+  set.migration = MIGRATION_NAME;
+
+  if (count % progressCount === 0) console.warn(`${count} ${user._id}`);
+  return await Tasks.Task.update({_id: user._id}, {$set: set}).exec();
+}
+
+module.exports = async function processUsers () {
+  let query = {
+    migration: {$ne: MIGRATION_NAME}
+  };
+
+  const fields = {
```

Comment:
this is going to select only _id and items (which is a user field, not a task one)